File: biosec/program/transaction.c
#include <stdio.h>
#include "account.h"
#include "ledger.h"

// ฝากเงิน
void deposit() {
    int id;
    double amount;
    printf("Enter account ID to deposit: ");
    scanf("%d", &id);

    Account *acc = find_account(id);
    if (!acc) {
        printf("Account not found!\n");
        return;
    }

    printf("Enter amount to deposit: ");
    scanf("%lf", &amount);
    if (amount <= 0) {
        printf("Invalid amount!\n");
        return;
    }

    acc->balance += amount;
    printf("Deposit successful! New balance: %.2f\n", acc->balance);

    // บันทึกลง ledger
    add_transaction(acc->id, "DEPOSIT", amount);
}

// ถอนเงิน
void withdraw() {
    int id;
    double amount;
    printf("Enter account ID to withdraw: ");
    scanf("%d", &id);

    Account *acc = find_account(id);
    if (!acc) {
        printf("Account not found!\n");
        return;
    }

    printf("Enter amount to withdraw: ");
    scanf("%lf", &amount);
    if (amount <= 0) {
        printf("Invalid amount!\n");
        return;
    }

    if (amount > acc->balance) {
        printf("Insufficient balance! Current balance: %.2f\n", acc->balance);
        return;
    }

    acc->balance -= amount;
    printf("Withdrawal successful! New balance: %.2f\n", acc->balance);

    // บันทึกลง ledger
    add_transaction(acc->id, "WITHDRAW", amount);
}

File: biosec/application/Transaction.java
public class Transaction {

    private int accountId;
    private String type; // "DEPOSIT" หรือ "WITHDRAW"
    private double amount;

    // Constructor
    public Transaction(int accountId, String type, double amount) {
        this.accountId = accountId;
        this.type = type;
        this.amount = amount;
    }

    // Getters
    public int getAccountId() {
        return accountId;
    }

    public String getType() {
        return type;
    }

    public double getAmount() {
        return amount;
    }

    // แสดงธุรกรรม
    public void display() {
        System.out.printf("AccountID: %d | Type: %s | Amount: %.2f\n", accountId, type, amount);
    }

    // ประมวลผลธุรกรรมกับบัญชี
    public void process(Account account) {
        if (!account.getIdEquals(accountId)) {
            System.out.println("Transaction account ID mismatch.");
            return;
        }

        if (type.equalsIgnoreCase("DEPOSIT")) {
            account.deposit(amount);
        } else if (type.equalsIgnoreCase("WITHDRAW")) {
            if (amount > account.getBalance()) {
                System.out.println("Insufficient balance for withdrawal.");
            } else {
                account.withdraw(amount);
            }
        } else {
            System.out.println("Invalid transaction type.");
        }
    }
}

